src/cli/commands/execute.rs

```diff
@@ -57,8 +63,39 @@ impl Display for ExecuteOutput {
 
 impl CommandExecution<ExecuteOutput> for ExecuteArgs {
 	fn exec(&self) -> Result<ExecuteOutput, String> {
-		let mut cairo_runner =
-			cairo_run(&self.program, "main", false, &HINT_EXECUTOR).map_err(|e| {
+		// Use cairo-compile binary in order to compile the .cairo file
+		let compiled_output: Output = Command::new("cairo-compile")
+			.args([&self.program])
+			.output()
+			.expect("cairo-compile command failed to start");
+
+		// Retrieve only the file name to create a clean compiled file name.
+		let program = self.program.display().to_string();
+		let filename: Vec<&str> = program.split('/').collect();
+		let filename = filename[filename.len() - 1].replace(".cairo", "");
```

Comment:
Use the Path rust lib. You are looking for file_stem: https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.file_stem